### ISSUE TITLE
Installation Script not working anymor

### ISSUE THREAD

**Issue by rene54321** [Race666/calimero-server#29]
Hi, could you please update the installation script to the latest version. The instllation of th used branch 2.5 M1 is not anymore wokring in my case. After manually changing the script to the newest branch 2.5.1 also an failure appears. I would be very happy about any help. Thanks René

**Comment by bmalinowsky** [comment]
You can check this version of the script which I updated for v2.5.1. I run it on armv7:

https://github.com/bmalinowsky/calimero-server/tree/release/2.5.1

**Comment by rene54321** [comment]
Thanks a lot the installation was working now. The only problem is that I receive many warnings like this:

"WARN calimero.server.gateway.knx-server - hop count 0, discard frame 1.0.2->10/1/3"

Do you have  an idea how to fix this?

best regards René

**Comment by bmalinowsky** [comment]
You quite likely have a network loop somewhere. Do you have a second KNXnet/IP router running on the same multicast address?

**Comment by rene54321** [comment]
Hi,

yes I have an Enertex KNX Secure Router in the same network but with an different static IP than the Raspberry.  Is it possible to configure the calimero only as IP-Interface and not an router?

best regards René

**Comment by bmalinowsky** [comment]
In _/etc/calimero/server-config.xml_, there is a setting 

`<serviceContainer activate="true" routing="true" ...` 
Set routing `false` and restart the systemd unit, or restart the raspberry.

Also, in the _calimero.sh_ script, there is a setting
`export KNX_ROUTING=true`
where you can set that behavior before running the installation script.

**Comment by rene54321** [comment]
Hi,

thanks a lot, this was working and this failure was not coming up again. Maybe you are Able to help me once more. I have an KNX TP Line with devices in the 1.1.x range and an KNX RF Line wirh devices in the 1.2.x range. The calimaro server has the address 1.1.0 and can send and receive data in this subnet. I also need to reach the RF Line 1.2.x which is not working. Folowing failure appears: 
" [knx-server subnet dispatcher] WARN calimero.server.gateway.knx-server - no subnet configured for destination 1.2.5 (received T_Connected "

Is it somehow possible to allow this via the calimaro config?

best regards René

**Comment by bmalinowsky** [comment]
There are 2 possibilities: 

1. change the server address to 1.0.0 and have the subnet be everything in range 1.x.y, for this the entry `<knxSubnet type="blah" medium="blah">blah</knxSubnet>` has to be a device which can address both lines 1.1. and 1.2. .
2. add a second service container (with address 1.2.0) in the config, and set the `<knxSubnet ...` line to point to a KNX RF device which is configured for your RF line.

Note, that if you change the server-config.xml manually, running the installation script again will overwrite your changes.
Also, the installation script only configures one service container (the one you use for TP1). So the second one you have to setup manually.

**Comment by bmalinowsky** [comment]
Did you try the suggestion I wrote above? The server gives that warning because with a configured address of 1.1.0 it won't forward into 1.2.x. This isn't surprising, it's an unrelated line.

**Comment by rene54321** [comment]
Hi,

do you mean like this:
    <knxAddress type="individual">1.1.128</knxAddress>
    <knxAddress type="individual">1.2.128</knxAddress>

best regards René

**Comment by bmalinowsky** [comment]
No, I mean the server address: the current entry `<knxAddress type="individual">1.1.0</knxAddress>` in the server config should be changed to contain the address `1.0.0` (approx. at line 8 in the file).

**Comment by bmalinowsky** [comment]
As long as you don't program the calimero server over ETS, reusing the address is fine.

The differences are:
* a TPUART doesn't care what you send out
* the USB device will quite likely insert its own device address as sender
* KNXD probably doesn't filter individual addresses
* calimero has an individual and (optional) group address filter 
 
The reason for the last point is that calimero supports > 1 service containers (i.e., KNX lines) and forwards messages between KNX lines. And a message with destination 1.2.x is not forwarded into a line which is configured as 1.1.0. You cannot have a device configured (in ETS, of course you can do that otherwise) behind a server/coupler which does not share the same address prefix. This would subvert the addressing scheme's purpose to route messages.

**Comment by Race666** [comment]
OK.
@bmalinowsky  Thx for your support!
